docs/docs/modules/builders/generating_transactions.mdx

```diff
@@ -0,0 +1,121 @@
+// pointer address
+const shelleyChangeAddress = CardanoWasm.Address.from_bech32("addr_test1gz2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzerspqgpsqe70et");
+
+// add output to the tx
+txBuilder.add_output(
+    CardanoWasm.TransactionOutput.new(
+    shelleyOutputAddress,
+    CardanoWasm.Value.new(CardanoWasm.BigNum.from_str('1000000'))    
+    ),
+);
+
+// set the time to live - the absolute slot value before the tx becomes invalid
+txBuilder.set_ttl(410021);
+
+// calculate the min fee required and send any change to an address
+txBuilder.add_change_if_needed(shelleyChangeAddress);
+
+// once the transaction is ready, we build it to get the tx body without witnesses
+const txBody = txBuilder.build();
+const txHash = CardanoWasm.hash_transaction(txBody);
+const witnesses = CardanoWasm.TransactionWitnessSet.new();
+
+// add keyhash witnesses
+const vkeyWitnesses = CardanoWasm.Vkeywitnesses.new();
+const vkeyWitness = CardanoWasm.make_vkey_witness(txHash, prvKey);
+vkeyWitnesses.add(vkeyWitness);
+witnesses.set_vkeys(vkeyWitnesses);
+
+// add bootstrap (Byron-era) witnesses
+const cip1852Account = CardanoWasm.Bip32PrivateKey.from_bech32('xprv1hretan5mml3tq2p0twkhq4tz4jvka7m2l94kfr6yghkyfar6m9wppc7h9unw6p65y23kakzct3695rs32z7vaw3r2lg9scmfj8ec5du3ufydu5yuquxcz24jlkjhsc9vsa4ufzge9s00fn398svhacse5su2awrw');
+const bootstrapWitnesses = CardanoWasm.BootstrapWitnesses.new();
+const bootstrapWitness = CardanoWasm.make_icarus_bootstrap_witness(
+    txHash,
+    byronAddress,
+    cip1852Account,
+);
+bootstrapWitnesses.add(bootstrapWitness);
+witnesses.set_bootstraps(bootstrapWitnesses);
+
+// create the finalized transaction with witnesses
+const transaction = CardanoWasm.Transaction.new(
+    txBody,
+    witnesses,
+    undefined, // transaction metadata
+);
+```
+
+## A note on fees
+
+Fees is Cardano Shelley are based directly on the size of the final encoded transaction. It is important to note that a transaction created by this library potentially can vary in size compared to one built with other tools. This is because transactions, as well as other Cardano Shelley structures, are encoded using [CBOR](https://cbor.io/) a binary JSON-like encoding. Due to arrays and maps allowing both definite or indefinite length encoding in the encoded transaction created by the library, the size can vary. This is because definite encoding consists of a tag containing the size of the array/map which can be 1 or more bytes long depending on the number of elements the size of the encoded structure, while indefinite length encoding consists of a 1 byte starting tag and after all elements are listed, a 1 byte ending tag. These variances should should only be a couple bytes and cardano-multiplatform-lib uses definite encoding which is the same length or smaller for any reasonable sized transaction.
```

Comment:
We do match encodings now (when creating via CBOR bytes) but will use definite encodings by default still so maybe this passage is still relevant. It could be worth mentioning this at least though. The "Cardano Shelley" wording as well. I know this was just pulled from the old docs to get the format in but we'll want to look through these moved-over things and make updates like this.